lua/lspconfig/robotframework_ls.lua

```diff
@@ -0,0 +1,24 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+local server_name = 'robotframework_ls'
+
+configs[server_name] = {
+  default_config = {
+    cmd = { 'robotframework_ls' },
+    filetypes = { 'robot' },
+    root_dir = function(fname)
+      return util.root_pattern '.git'(fname) or util.path.dirname(fname)
```

Comment:
We won't support `util.path.dirname(fname)` fallbacks until scratch server support has been implemented. However, I had a brief look at the robotframework-lsp and it seems like you'd want these root markers:

```python
    for directory in (common_base, *common_base.parents):
        if (directory / ".git").exists():
            return directory


        if (directory / "robotidy.toml").is_file():
            return directory


        if (directory / "pyproject.toml").is_file():
            return directory
```

https://github.com/robocorp/robotframework-lsp/blob/e2ea48e488840f592b20057f6049231d9cb05a01/robocorp-python-ls-core/src/robocorp_ls_core/libs/robotidy_lib/robotidy/files.py#L36-L44

Reply:
```suggestion
      return util.root_pattern('pyproject.toml', 'robotidy.toml') or util.find_git_ancestor(fname)
```